edit-post/api/more-menu-item.js

```diff
@@ -0,0 +1,91 @@
+/* eslint no-console: [ 'error', { allow: [ 'error' ] } ] */
+
+/**
+ * Wordpress dependencies
+ */
+import { applyFilters } from '@wordpress/hooks';
+import { validateNamespacedId } from '@wordpress/utils';
+
+/**
+ * Internal dependencies
+ */
+import { isString } from 'lodash';
+import { activateSidebar } from './sidebar';
+
+const menuItems = {};
+
+/**
+ * Registers a plugin under the more menu.
+ *
+ * @param {string}       menuItemId      The unique identifier of the plugin. Should be in
+ *                                       `[namespace]/[name]` format.
+ * @param {Object}       settings        The settings for this menu item.
+ * @param {string}       settings.title  The name to show in the settings menu.
+ * @param {func}         settings.target The registered plugin that should be activated.
+ * @param {ReactElement} [settings.icon] SVG React Component.
```

Comment:
Components and elements are not synonymous:

https://reactjs.org/blog/2015/12/18/react-components-elements-and-instances.html

Further, the abstraction of `wp.element` should be respected here. These isn't strictly a React Component, it's a "WordPress Element Component", or however we'd intend to express this (`WPComponent`).